src/main/java/org/spongepowered/common/world/border/PlayerOwnBorderListener.java

```diff
@@ -0,0 +1,81 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.world.border;
+
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.network.Packet;
+import net.minecraft.network.play.server.SPacketWorldBorder;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.world.border.IBorderListener;
+import net.minecraft.world.border.WorldBorder;
+import org.spongepowered.api.entity.living.player.Player;
+
+public class PlayerOwnBorderListener implements IBorderListener {
+
+    private final EntityPlayerMP player;
+
+    public PlayerOwnBorderListener(EntityPlayerMP player) {
+        this.player = player;
+    }
+
+    @Override
+    public void onSizeChanged(WorldBorder border, double newSize) {
+        sendBorderPacket(new SPacketWorldBorder(border, SPacketWorldBorder.Action.SET_SIZE), border);
+    }
+
+    @Override
+    public void onTransitionStarted(WorldBorder border, double oldSize, double newSize, long time) {
+        sendBorderPacket(new SPacketWorldBorder(border, SPacketWorldBorder.Action.LERP_SIZE), border);
+    }
+
+    @Override
+    public void onCenterChanged(WorldBorder border, double x, double z) {
+        sendBorderPacket(new SPacketWorldBorder(border, SPacketWorldBorder.Action.SET_CENTER), border);
+    }
+
+    @Override
+    public void onWarningTimeChanged(WorldBorder border, int newTime) {
+        sendBorderPacket(new SPacketWorldBorder(border, SPacketWorldBorder.Action.SET_WARNING_TIME), border);
+    }
+
+    @Override
+    public void onWarningDistanceChanged(WorldBorder border, int newDistance) {
+        sendBorderPacket(new SPacketWorldBorder(border, SPacketWorldBorder.Action.SET_WARNING_BLOCKS), border);
+    }
+
+    @Override
+    public void onDamageAmountChanged(WorldBorder border, double newAmount) {
+    }
+
+    @Override
+    public void onDamageBufferChanged(WorldBorder border, double newSize) {
+    }
+
+    private void sendBorderPacket(Packet<?> packet, WorldBorder border) {
+        if (((Player) player).getWorldBorder().orElse(null) == border && border != null) {
```

Comment:
This will work but I'm worried that this class will leak the `player`  instance if the `WorldBorder` persists after the player leaves the game (because you never remove the listener from the world border). I think the best solution for this is to track the current players in the `WorldBorder` (e.g. in a list) and remove the players when they leave the server.